microcloud/test/includes/microcloud.sh

```diff
@@ -966,14 +966,13 @@ setup_system() {
   # Sleep some time so the snaps are fully set up.
   sleep 3
 
-
-  lxc stop "${name}"
-
-  lxc snapshot "${name}" snap0
-
-  lxc start "${name}"
-
-  lxd_wait_vm "${name}"
+  # Create a snapshot so we can restore to this point.
+  if [ "${SNAPSHOT_RESTORE}" = 1 ]; then
+    lxc stop "${name}"
```

Comment:
lxc stop --force?

Reply:
We're snapshotting the VM here, so if we `lxc stop -f`, that makes the resulting snapshot less deterministic as we just force quit everything.

Reply:
i think in the lxd vm tests we execute a `sync` call and then force stop to ensure it runs more reliably

https://github.com/canonical/lxd-ci/blob/main/tests/storage-vm#L83-L86

Reply:
We run that function/take that snapshot only once so I think it's safer to play it safe and that the speed hit which should be small as those system were freshly created to there is little risk for them to hang during stop.